r/R/array.R

```diff
@@ -322,6 +322,26 @@ as_arrow_array.data.frame <- function(x, ..., type = NULL) {
   }
 }
 
+#' @export
+as_arrow_array.vctrs_list_of <- function(x, ..., type = NULL) {
+  type <- type %||% infer_type(x)
```

Comment:
Any reason not to do this?

```suggestion
as_arrow_array.vctrs_list_of <- function(x, ..., type = infer_type(x)) {
```

Reply:
If we still want to support passing explicit `NULL`, the existing implementation seems better.